core/src/main/scala/kafka/coordinator/transaction/TransactionManager.scala

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.coordinator.transaction
+
+import java.nio.ByteBuffer
+import java.util.Properties
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.locks.ReentrantLock
+
+import kafka.common.Topic
+import kafka.log.LogConfig
+import kafka.server.ReplicaManager
+import kafka.utils.CoreUtils.inLock
+import kafka.utils.{Logging, Pool, Scheduler, ZkUtils}
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.record.{FileRecords, MemoryRecords}
+import org.apache.kafka.common.utils.{Time, Utils}
+
+import scala.collection.mutable
+import scala.collection.JavaConverters._
+
+/*
+ * Transaction manager is part of the transaction coordinator, it manages:
+ *
+ * 1. the transaction log, which is a special internal topic.
+ * 2. the pid metadata including its ongoing transaction status.
+ * 3. the background expiration of the transaction as well as the transactional id to pid mapping.
+ *
+ */
+class TransactionManager(brokerId: Int,
```

Comment:
Hmm... `TransactionManager` sounds a lot like `TransactionCoordinator`. It would be nice to keep the scope of responsibility a bit clearer and it seems this is mostly concerned with managing the storage of the transaction state.